lib/ansible/module_utils/network/ftd/common.py

```diff
@@ -65,7 +65,7 @@ def construct_ansible_facts(response, params):
         response_body = response['items'] if 'items' in response else response
         if params.get('register_as'):
             facts[params['register_as']] = response_body
-        elif response_body.get('name') and response_body.get('type'):
+        elif type(response_body) is dict and response_body.get('name') and response_body.get('type'):
```

Comment:
note, instead of using type for this, isinstance is the preferred method of doing things.
```python
from ansible.module_utils.common._collections_compat import Mapping
[...]
        elif isinstance(response_body, Mapping) and response_body.get('name') and response_body.get('type'):
```

Reply:
@annikulin plz implement the change suggestion made in `devel`